File: practice/data_structures/linked_lists/delete_a_node/delete_a_node.py
"""
 Delete Node at a given position in a linked list
 Node is defined as
 
 class Node(object):
 
   def __init__(self, data=None, next_node=None):
       self.data = data
       self.next = next_node

 return back the head of the linked list in the below method. 
"""

def Delete(i_head, i_position):
    if i_position == 0:
        i_head = i_head.next
        return i_head
    cp = i_head
    while i_position > 1:
        cp = cp.next
        i_position -= 1
    cp.next = cp.next.next
    return i_head

File: practice/data_structures/linked_lists/delete_a_node/delete_a_node.cpp
/*
  Delete Node at a given position in a linked list 
  Node is defined as 
  struct Node
  {
     int data;
     struct Node *next;
  }
*/

Node * Delete(Node * i_head, int i_position)
{
    if (i_position == 0)
    {
        i_head = i_head->next;
        return i_head;
    }
    
    Node * node = i_head;
    while (i_position > 1)
    {
        node = node->next;
        i_position--;
    }
    
    node->next = node->next->next;
    return i_head;
}

